### PR TITLE
Mubarak's Pull Request

### PR DESCRIPTION
**Pull request by Creatorrr1**

### CODE REVIEW

File: src/cohort.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    for (let i = 0; i < this.students.length; i++) {
+      if (this.students[i].studentId === studentID) {
+        this.students.splice(i, 1);
+        return this.students;
+      }
+    }
+
+    return existError;
+  }
```
Comment (Dan2024) [inline src/cohort.js:38]: is there a reason you declared the error message as a variable on line 18? if not you might be able to save a few lines of code by deleting line 18 and instead on line 27 typing `return "Student is not in the register"` 👍

File: src/cohortManager.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+addStudentToCohort (cohortName, student) {
+      //   find the cohort with cohortName 
+    //   if not found return error
+    // if found return cohort.add(student)
+    const cohort = this.foundCohort(cohortName)
+    if (cohort !== false ) {
+        return cohort.add(student)
+    } else {
+        return false
+    }
```
Comment (Dan2024) [inline src/cohortManager.js]: nice use of pseudo code and calling other functions you created elsewhere in your code

File: src/cohortManager.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      }
+    }
+    return false
+  }
+
+  remove(cohortN) {
+    const errorMessage = "cohort not found!";
+    for (let i = 0; i < this.cohortClasses.length; i++) {
+      if (cohortN === this.cohortClasses[i].cohortName) {
+        return this.cohortClasses.splice(i, 1)[0];
```
Comment (Dan2024) [inline src/cohortManager.js:68]: what's the [0] at the end of this line doing? I've seen other people use it a few times too

File: src/cohortManager.js [hunk 4]
```diff
@@ -5,12 +5,28 @@ class CohortManager {
        this.cohortClasses = [] 
     }
 
-    add(cohort) {
-        this.cohortClasses.push(cohort)
+    add(cohortN) {
+        this.cohortClasses.push(cohortN)
```
Comment (Hoolio333) [inline src/cohortManager.js]: Simple but effective - the best way

File: src/cohortManager.js [hunk 5]
```diff
@@ -5,12 +5,28 @@ class CohortManager {
...
     }
 
-    remove() {
-        
+    errorMessage = 'cohort not found!'
+
+    foundCohort(cohortN) {
```
Comment (Hoolio333) [inline src/cohortManager.js]: Maybe a use a better name than cohortN.  Not easy to know if the N stands for name? number? etc without looking elsewhere.

File: domain-model.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
+Pseudocode | Objects  | Methods |Messages / Properties | Output
+---- |---- | ---- | ---- | ----
+|| Cohort (class)||this.classes @array||
+|||createCohort(cohortNum)||cohort pushed into this.classes @array
+|||removeCohort()| cohort removed by name but first searched | cohort removed from this.classes @array
+|||addStudent()|add student to cohort|
+|||removeStudent()|remove student from cohort|
+||Students()||studentID@number,firstName&LastName @string|
```
Comment (Hoolio333) [inline domain-model.md]: Hard to make domain models easy to read & follow.  Maybe use a spreadsheet next time & attach the image.

File: src/cohortManager.js [hunk 7]
```diff
@@ -1,33 +1,61 @@
...
+      }
+    }
+    return false
+  }
+
+  remove(cohortN) {
+    const errorMessage = "cohort not found!";
+    for (let i = 0; i < this.cohortClasses.length; i++) {
+      if (cohortN === this.cohortClasses[i].cohortName) {
+        return this.cohortClasses.splice(i, 1)[0];
```
Comment (Hoolio333) [inline src/cohortManager.js:68]: As Dan said, not sure what the [0] is doing?

**Pull request closed without merging**